src/model/Course.java

```diff
@@ -12,36 +12,949 @@
+	 *                            {@link CourseCategory} enum.
+	 * @param schedule            Course schedule for a semester of the course.
+	 * @param courseBook          General book of the course.
+	 * @param suggestedReading    List of suggested readings for more detailed
+	 *                            information.
+	 * @param evaluationCriterias Evaluation criterias of the course to calculate
+	 *                            the total grade. // * @param workloadTable All
+	 *                            workload that a course require in one semester.
+	 * @param courseCompetencies  Competencies of the course with related learning
+	 *                            outcomes.
+	 */
+	public Course(String courseName, String code, LinkedHashSet<String> prerequisites, Language courseLang,
+			SemesterOptions semester, int theoreticalHour, int labHour, CourseType type, Lecturer coordinator,
+			LinkedHashSet<Lecturer> lecturers, LinkedHashSet<Lecturer> assistants, String courseObjective,
+			LinkedHashSet<String> learningOutcomes, String courseDescription, CourseCategory category,
+			LinkedHashSet<WeeklySubject> schedule, CourseBook courseBook, LinkedHashSet<CourseBook> suggestedReading,
+			LinkedHashSet<EvaluationCriteria> evaluationCriterias, LinkedHashSet<SemesterActivity> workloadTable,
+			LinkedHashSet<CourseCompetency> courseCompetencies) {
+		this.courseName = courseName;
+		this.code = code;
+		this.creationDate = LocalDateTime.now();
+		this.prerequisites = new ArrayList<>(prerequisites);
+		this.courseLang = courseLang;
+		this.labHour = labHour;
+		this.theoreticalHour = theoreticalHour;
+		this.localCredit = calculateLocalCredit();
+		this.ects = calculateECTS();
+		this.type = type;
+		this.courseCordinator = coordinator;
+		this.courseLecturers = new ArrayList<>(lecturers);
+		this.assistants = new ArrayList<>(assistants);
+		this.courseObjective = courseObjective;
+		this.courseDescription = courseDescription;
+		this.learningOutcomes = new ArrayList<>(learningOutcomes);
+		this.courseCategory = category;
+		this.courseSchedule = new ArrayList<>(schedule);
+		this.courseTextBook = courseBook;
+		this.suggestedReading = new ArrayList<>(suggestedReading);
+		this.evaluationCriterias = new ArrayList<>(evaluationCriterias);
+		this.workloadTable = new ArrayList<>(workloadTable);
+		this.courseCompetencies = new ArrayList<>(courseCompetencies);
+	}
 	
+	//FIXME add a default constructor for newly created courses without a sufficient information supplied.
+
+	// Method(s)
+	/**
+	 * Add a prerequisite to the course.
+	 * 
+	 * @param courseCode Code of course that will be added as a prerequisites.
+	 * @return {@code true} if the new course added successfully to the
+	 *         {@link #prerequisites} list, return {@code false} otherwise.
+	 */
+	public boolean addPrerequisite(String courseCode) {
+		if (doesContain(prerequisites, courseCode)) {
+			return false;
+		} else {
+			return prerequisites.add(courseCode);
```

Comment:
Burada her ders için önkoşul bir dersi almak olmayabiliyor, mesela bitirme projelerinin önkoşulu 4. sınıf olmak, yani aslında courseCode yerine istediğimizi yazabiliriz değil mi?

Reply:
Böyle bir durumdan haberim yoktu inceleyip belirtmen gerçekten çok iyi oldu. Buna uygun olacak şekilde düzenlemeleri yapacağım.